### PR TITLE
feat: add todos_overview bloc

### PR DESCRIPTION
**Pull request by Limbou**
## Description

Add todos_overview bloc and models

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/todos_overview/bloc/todos_overview_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:todo_app/todos_overview/models/models.dart';
```
Comment (orestesgaolin) [inline lib/todos_overview/bloc/todos_overview_bloc.dart:3]:
```suggestion
import 'package:todo_app/todos_overview/todos_overview.dart';
```

We prefer having top level barrel files for each feature, this way we don't use inner export files. In the future we may have some more tooling to verify that

File: test/todos_overview/bloc/todos_overview_bloc_test.dart [hunk 2]
```diff
@@ -0,0 +1,385 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:todo_app/todos_overview/bloc/todos_overview_bloc.dart';
+import 'package:todo_app/todos_overview/models/models.dart';
```
Comment (orestesgaolin) [inline test/todos_overview/bloc/todos_overview_bloc_test.dart:5]:
```suggestion
import 'package:todo_app/todos_overview/todos_overview.dart';
```

File: test/todos_overview/bloc/todos_overview_bloc_test.dart [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+          const TodosOverviewState(status: TodosOverviewStatus.failure),
+        ],
+      );
+
+      test('event object props is empty', () {
+        expect(
+          const TodosOverviewSubscriptionRequested().props,
+          isEmpty,
+        );
+      });
```
Comment (orestesgaolin) [inline test/todos_overview/bloc/todos_overview_bloc_test.dart:110]: You can replace this test with checking for value equality. Make sure to use non const constructors (e.g. add ignore for the file for the lint rule). This way by comparing two non-const objects the equatable will access props.

Comment (orestesgaolin) [reply test/todos_overview/bloc/todos_overview_bloc_test.dart:110]: btw tests for `TodosOverviewSubscriptionRequested` should be in the dedicated todos_overview_event_test or similar

File: test/todos_overview/bloc/todos_overview_bloc_test.dart [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+        },
+        errors: () => [isA<AssertionError>()],
+      );
+
+      test('props are empty', () {
+        expect(
+          const TodosOverviewUndoDeletionRequested().props,
+          isEmpty,
+        );
+      });
```
Comment (orestesgaolin) [inline test/todos_overview/bloc/todos_overview_bloc_test.dart:258]: same comment as above, testing for props being empty is a bit tedious and low level

File: test/todos_overview/bloc/todos_overview_bloc_test.dart [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+          ],
+        ),
+        act: (bloc) {
+          bloc.add(
+            const TodosOverviewToggleAllRequested(),
+          );
+        },
+        verify: (_) {
+          verify(
+            () => todosRepository.completeAll(isCompleted: true),
```
Comment (orestesgaolin) [inline test/todos_overview/bloc/todos_overview_bloc_test.dart:336]: we can talk this on Slack but I'm not sure if I understand the API right away; if I want to "complete all" why do I need to pass additional argument?

File: test/todos_overview/models/todos_view_filter_test.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// ignore_for_file: prefer_const_constructors
+import 'package:test/test.dart';
+import 'package:todo_app/todos_overview/models/models.dart';
+import 'package:todos_repository/todos_repository.dart';
```
Comment (orestesgaolin) [inline test/todos_overview/models/todos_view_filter_test.dart:4]:
```suggestion
import 'package:todo_app/todos_overview/todos_overview.dart';
import 'package:todos_repository/todos_repository.dart';
```

**Review by orestesgaolin — APPROVED** [review]
Few remarks about testing but in general really good job

File: test/todos_overview/bloc/todos_overview_bloc_test.dart (hunk 3 above)
Comment (Limbou) [reply test/todos_overview/bloc/todos_overview_bloc_test.dart:110]: Oh, I didn't know about it. What would be the reason for two non-const objects to be compared using propos, but two const objects not?

**Pull request merged**